pkg/agents/default/baseagent.go

```diff
@@ -6,15 +6,21 @@ import (
 
 	tower "github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
```

Comment:
```suggestion
	"github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
```

You will need to change the package name in tower to `tower`. 
i.e. the first line in `tower.go` should be `package tower`, not `package Tower`.
Capital letters in package names are bad practice in Go.